File: CS/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using DevExpress.XtraBars;

namespace WindowsApplication1
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        Control popupControl;

        private void barButtonItem1_ItemClick(object sender, DevExpress.XtraBars.ItemClickEventArgs e)
        {
            Console.WriteLine(e.Link.LinkedObject);
            if (e.Link.LinkedObject == popupMenu1)
                MessageBox.Show(string.Format("The context menu was invoked for '{0}'.", popupControl));
        }

        private void Form1_Load(object sender, EventArgs e)
        {
            ribbonControl1.Manager.QueryShowPopupMenu += ShowPopupMenuHandler;
        }        

        private void ShowPopupMenuHandler(object sender, QueryShowPopupMenuEventArgs e)
        {
            Console.WriteLine(e.Control);
            popupControl = e.Control;
        }

        private void popupMenu1_CloseUp(object sender, EventArgs e)
        {
            popupControl = null;
        }
    }
}

File: VB/Form1.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms
Imports DevExpress.XtraBars

Namespace WindowsApplication1
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private popupControl As Control

		Private Sub barButtonItem1_ItemClick(ByVal sender As Object, ByVal e As DevExpress.XtraBars.ItemClickEventArgs) Handles barButtonItem1.ItemClick
			Console.WriteLine(e.Link.LinkedObject)
			If e.Link.LinkedObject Is popupMenu1 Then
				MessageBox.Show(String.Format("The context menu was invoked for '{0}'.", popupControl))
			End If
		End Sub

		Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs) Handles MyBase.Load
			AddHandler ribbonControl1.Manager.QueryShowPopupMenu, AddressOf ShowPopupMenuHandler
		End Sub

		Private Sub ShowPopupMenuHandler(ByVal sender As Object, ByVal e As QueryShowPopupMenuEventArgs)
			Console.WriteLine(e.Control)
			popupControl = e.Control
		End Sub

		Private Sub popupMenu1_CloseUp(ByVal sender As Object, ByVal e As EventArgs) Handles popupMenu1.CloseUp
			popupControl = Nothing
		End Sub
	End Class
End Namespace